apps/frontend/src/app/5_pages/ConvertPage/ConvertPage.tsx

```diff
@@ -114,10 +123,19 @@ const ConvertPage: FC = () => {
     useState<SupportedTokens>();
 
   useEffect(() => {
-    smartRouter
-      .getEntries()
-      .then(tokens => tokensToOptions(tokens, setTokenOptions));
-  }, []);
+    smartRouter.getEntries().then(tokens => {
+      tokensToOptions(tokens, options => {
+        if (!hasMyntBalance) {
+          const filteredOptions = options.filter(
+            option => option.value !== SupportedTokens.mynt,
+          );
+          setTokenOptions(filteredOptions);
+        } else {
+          setTokenOptions(options);
+        }
+      });
+    });
+  }, [sourceToken, hasMyntBalance]);
```

Comment:
Source tokens are being refetched each time sourceToken or mynt balance flag changes. We only need to retrieve sources once.

Make a `useMemo` which filters MYNT token in/out from `tokenOptions` when `hasMyntBalance` is changed instead.

```suggestion
    smartRouter
      .getEntries()
      .then(tokens => tokensToOptions(tokens, setTokenOptions));
  }, []);
```